contracts/utils/math/Math.sol

```diff
@@ -218,6 +218,31 @@ library Math {
         return result;
     }
 
+    /**
+     * @notice Calculate the inverse of a number in Z/nZ.
+     *
+     * If p is a prime, then we Z/nZ is a field commonly noted Fp. In that case all elements are inversible, expect 0.
+     * If p is not a prime, then Z/nZ is not a field, and some elements might not be inversible.
```

Comment:
If I understood correctly, this is a field, but it's just not a finite field. If I'm wrong, would you mind explaining the difference?